Mac build failing

I didn't find any bug to track this but I can't build on Mac OSX

$ cmake .
-- Could NOT find MMAL (missing:  MMAL_LIBRARIES MMAL_INCLUDE_DIRS) 
-- Could NOT find MYSQL (missing:  MYSQL_LIBRARY MYSQL_INCLUDE_DIR) 
-- Could NOT find PostgreSQL (missing:  PostgreSQL_INCLUDE_DIR PostgreSQL_TYPE_INCLUDE_DIR) 
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.14.0
-- Checking for modules 'libavutil;libavformat;libavcodec;libswscale'
--   Found libavutil, version 55.34.101
--   Found libavformat, version 57.56.101
--   Found libavcodec, version 57.64.101
--   Found libswscale, version 4.2.100
-- Configuration: 
--   Package Name:           motion
--   Package Version:        4.0.1+gitede7a0b
--   Build System:           Darwin
--   Config Dir:             /usr/local/etc
--   Bin Dir:                /usr/local/bin
--   JPEG Include Dir:       /usr/local/include
--   WITH_V4L2:              ON
--   HAVE_V4L2:              OFF
--   WITH_BKTR:              OFF
--   HAVE_BKTR:              OFF
--   MYSQL:   OFF Include Dir: MYSQL_INCLUDE_DIR-NOTFOUND
--   SQLITE3: OFF Include Dir: 
--   PGSQL:   OFF Include Dir: 
--   FFMPEG:  ON Include Dir: /usr/local/Cellar/ffmpeg/3.2.4/include
--   MMAL:    OFF
--   CFLAGS:   -std=gnu99 -Wall
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/mike/Projects/motion/motion

$ make
[  4%] Building C object CMakeFiles/motion.dir/conf.c.o
In file included from /Users/mike/Projects/motion/motion/conf.c:30:
/Users/mike/Projects/motion/motion/motion.h:51:10: fatal error: 'pthread_np.h' file not found
#include <pthread_np.h>
         ^
1 error generated.
make[2]: *** [CMakeFiles/motion.dir/conf.c.o] Error 1
make[1]: *** [CMakeFiles/motion.dir/all] Error 2
make: *** [all] Error 2

looks like commenting it out works, it builds, haven't tried it out yet:

```
make
Scanning dependencies of target motion
[  4%] Building C object CMakeFiles/motion.dir/conf.c.o
[  8%] Building C object CMakeFiles/motion.dir/motion.c.o
[ 12%] Building C object CMakeFiles/motion.dir/alg.c.o
[ 16%] Building C object CMakeFiles/motion.dir/draw.c.o
[ 20%] Building C object CMakeFiles/motion.dir/event.c.o
[ 25%] Building C object CMakeFiles/motion.dir/ffmpeg.c.o
/Users/mike/Projects/motion/motion/ffmpeg.c:304:23: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        avcodec_close(AVSTREAM_CODEC_PTR(ffmpeg->video_st));
                      ^
/Users/mike/Projects/motion/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/motion/motion/ffmpeg.c:343:17: warning: 'avcodec_encode_video2' is deprecated [-Wdeprecated-declarations]
        retcd = avcodec_encode_video2(AVSTREAM_CODEC_PTR(ffmpeg->video_st),
                ^
/usr/local/include/libavcodec/avcodec.h:5323:5: note: 'avcodec_encode_video2' has been explicitly marked deprecated here
int avcodec_encode_video2(AVCodecContext *avctx, AVPacket *avpkt,
    ^
/Users/mike/Projects/motion/motion/ffmpeg.c:343:39: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        retcd = avcodec_encode_video2(AVSTREAM_CODEC_PTR(ffmpeg->video_st),
                                      ^
/Users/mike/Projects/motion/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/motion/motion/ffmpeg.c:669:25: warning: 'codec' is deprecated [-Wdeprecated-declarations]
    ffmpeg->c     = c = AVSTREAM_CODEC_PTR(ffmpeg->video_st);
                        ^
/Users/mike/Projects/motion/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/motion/motion/ffmpeg.c:861:23: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        avcodec_close(AVSTREAM_CODEC_PTR(ffmpeg->video_st));
                      ^
/Users/mike/Projects/motion/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
5 warnings generated.
[ 29%] Building C object CMakeFiles/motion.dir/jpegutils.c.o
[ 33%] Building C object CMakeFiles/motion.dir/logger.c.o
[ 37%] Building C object CMakeFiles/motion.dir/md5.c.o
[ 41%] Building C object CMakeFiles/motion.dir/netcam.c.o
[ 45%] Building C object CMakeFiles/motion.dir/netcam_ftp.c.o
[ 50%] Building C object CMakeFiles/motion.dir/netcam_jpeg.c.o
[ 54%] Building C object CMakeFiles/motion.dir/netcam_rtsp.c.o
/Users/mike/Projects/motion/motion/netcam_rtsp.c:102:13: warning: 'avcodec_decode_video2' is deprecated [-Wdeprecated-declarations]
    retcd = avcodec_decode_video2(cc, frame, &check, packet);
            ^
/usr/local/include/libavcodec/avcodec.h:4812:5: note: 'avcodec_decode_video2' has been explicitly marked deprecated here
int avcodec_decode_video2(AVCodecContext *avctx, AVFrame *picture,
    ^
/Users/mike/Projects/motion/motion/netcam_rtsp.c:160:19: warning: 'codec' is deprecated [-Wdeprecated-declarations]
    dec_ctx = st->codec;
                  ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/motion/motion/netcam_rtsp.c:490:108: warning: 'codec' is deprecated [-Wdeprecated-declarations]
    netcam->rtsp->codec_context = netcam->rtsp->format_context->streams[netcam->rtsp->video_stream_index]->codec;
                                                                                                           ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
3 warnings generated.
[ 58%] Building C object CMakeFiles/motion.dir/netcam_wget.c.o
[ 62%] Building C object CMakeFiles/motion.dir/picture.c.o
[ 66%] Building C object CMakeFiles/motion.dir/rotate.c.o
[ 70%] Building C object CMakeFiles/motion.dir/stream.c.o
[ 75%] Building C object CMakeFiles/motion.dir/track.c.o
[ 79%] Building C object CMakeFiles/motion.dir/video_loopback.c.o
[ 83%] Building C object CMakeFiles/motion.dir/webhttpd.c.o
[ 87%] Building C object CMakeFiles/motion.dir/video_v4l2.c.o
[ 91%] Building C object CMakeFiles/motion.dir/video_common.c.o
[ 95%] Building C object CMakeFiles/motion.dir/video_bktr.c.o
[100%] Linking C executable motion
[100%] Built target motion
```

--- a/motion.h
+++ b/motion.h
@@ -47,7 +47,8 @@
 #include <stdint.h>
 #include <pthread.h>
 
-#if defined(BSD)
+#ifdef __FreeBSD__
+//It is unknown whether it is safe to put this into the if/else below on Apple and BSD
 #include <pthread_np.h>
 #endif
 

